openpype/modules/shotgrid/lib/settings.py

```diff
@@ -0,0 +1,41 @@
+import os
+
+from pymongo import MongoClient
+from openpype.api import get_system_settings, get_project_settings
+from openpype.modules.shotgrid.lib.const import MODULE_NAME
+from openpype.modules.shotgrid.lib.tools import memoize
+
+
+def get_project_list():
+    mongo_url = os.getenv("OPENPYPE_MONGO")
+    client = MongoClient(mongo_url)
+    db = client['avalon']
+    return db.list_collection_names()
+
+
+@memoize
```

Comment:
Function `memoize` is used to cache settings? You'll loose dynamic approach of settings which gives a lot of variability to the product. If this is called so often it's slowing down process so match, it is a question why it's called so ofter?

Reply:
After the first review, your colleague prevented us from using such dynamic approach and strongly recommended using some cache mechanics. So I'm quite perplexed, what is supposed to be used?

Reply:
I guess it was not meant globaly but for the specific purpose where it was used.

Reply:
> I guess it was not meant globaly but for the specific purpose where it was used.

No, it was about the portion of the code where memoization was finally at use. You can find more specific details in our above [conversation](https://github.com/pypeclub/OpenPype/pull/2921#discussion_r834145139) with @BigRoy. After the memoization was added @BigRoy seemed to [approve](https://github.com/pypeclub/OpenPype/pull/2921#pullrequestreview-921495465) changes. I quote: "The code updates look good to me". We learned as well from previous conversation with your team that direct querying of mongodb should be avoided because of some use cases where users have poor connectivity to cheap mongodb instances. (@ClementHector) So I'm confused, should we avoid caching(memoization, lru, global var) or not.

Reply:
@victorbartel Just to mention it: Sorry if my approval might have put this in the wrong direction. (Thanks for tagging me btw!)

Having cached data can be very helpful for performance (at the expense of remaining in memory of course) but I do agree with @iLLiCiTiT that we should avoid it if we could.